proto/structured.proto

```diff
@@ -104,3 +105,237 @@ message CreateTableResponse {
+  // CmdID is optionally specified for request idempotence
+  // (i.e. replay protection).
+  optional ClientCmdID cmd_id = 4 [(gogoproto.nullable) = false, (gogoproto.customname) = "CmdID"];
+  // User is the originating user. Used to lookup priority when
+  // scheduling queued operations at target node.
+  optional string user = 5 [(gogoproto.nullable) = false];
+  // UserPriority specifies priority multiple for non-transactional
+  // commands. This value should be a positive integer [1, 2^31-1).
+  // It's properly viewed as a multiple for how likely this
+  // transaction will be to prevail if a write conflict occurs.
+  // Commands with UserPriority=100 will be 100x less likely to be
+  // aborted as conflicting transactions or non-transactional commands
+  // with UserPriority=1. This value is ignored if Txn is
+  // specified. If neither this value nor Txn is specified, the value
+  // defaults to 1.
+  optional int32 user_priority = 6 [default = 1];
+  // Txn is set non-nil if a transaction is underway. To start a txn,
+  // the first request should set this field to non-nil with name and
+  // isolation level set as desired. The response will contain the
+  // fully-initialized transaction with txn ID, priority, initial
+  // timestamp, and maximum timestamp. The transaction returned in a
+  // response is feed to the next request.
+  optional Transaction txn = 7;
+  // ReadConsistency specifies the consistency for read
+  // operations. The default is CONSISTENT. This value is ignored for
+  // write operations.
+  optional ReadConsistencyType read_consistency = 8 [(gogoproto.nullable) = false];
+}
+
+// TableResponseHeader is returned with every structured API response.
+message TableResponseHeader {
+  // Error is non-nil if an error occurred.
+  optional Error error = 1;
+  // Timestamp specifies time at which read or write actually was
+  // performed. In the case of both reads and writes, if the timestamp
+  // supplied to the request was 0, the wall time of the node
+  // servicing the request will be set here. Additionally, in the case
+  // of writes, this value may be increased from the timestamp passed
+  // with the RequestHeader if the key being written was either read
+  // or written more recently.
+  //
+  // TODO(vivek): Is this only relevant in the non-transactional case?
+  optional Timestamp timestamp = 2 [(gogoproto.nullable) = false];
+  // Transaction is non-nil if the request specified a non-nil
+  // transaction. The transaction timestamp and/or priority may have
+  // been updated, depending on the outcome of the request.
+  optional Transaction txn = 3;
+}
+
+// A ColumnCollection represents a collection of columns as name:value pairs.
+message ColumnCollection {
+  message ColumnNameValue {
+    // Column name.
+    optional string name = 1;
+    // A Nil Value should be represented as Nil Bytes, indicating
+    // non-existence of the value. 
+    optional Value value = 2 [(gogoproto.nullable) = false];
```

Comment:
We return a ValueCollection in the response to Get/Scan, where the semantics is that the Values are returned in the same order as the requested columns. So the Values in ValueCollection are non-nullable in the sense that the same number of values have to be returned as the columns requested. I could change this to be nullable since it's attached to a name in this message but then it's existential-semantics is different from the Value in ValueCollection and that's a bit disconcerting.